docs/parse_argparse.py

```diff
@@ -0,0 +1,315 @@
+                # the way this `if` statements works is: os.walk will keep iterating through a directory recursively (going into sub-sub-directories), we just need its first output
+                # of the directories within the root, after that it goes through the sub-sub-directories and lists their directories which we don't want,
+                # meanwhile path_main_file is constant, thus we can skip recording sub-subdirectories if we tell it that, this path has alreasy been recorded.
+                # remember path_main_file is remaining constant for one dir_path
+                if path_recorded == path_main_file:
+                    continue
+                # this counts the number of sub-directories each directory has
+                # true only for `common` (as of April 24, 2020)
+                if len(dirnames) > 0:
+                    for dirname in dirnames:
+                        if "__" in dirname:  # to prevent __pychache__ folders from being considered
+                            continue
+                        dir_main = dirname + "_main"  # external_main
+                        path_recorded = path_main_file
+                        lines = f"\n\t {dirname}/{dir_main}"
+                        outlines += lines
+            fh.write(outlines)
+    # Opens each DirectoryName_main.rst and add the name of the scripts in that directory
+    with open(path_main_file, "a") as fh:
+        fname = os.path.basename(fpath)
+        # extract the "kmers" of kmers.py
+        basename = os.path.splitext(fname)[0]
+        fh.write(f"\n\t {basename}")
+
+
+def link_write_usage():
+    """
+    Creates usage.rst and writes the directories in its toctree
+    """
+    if not os.path.exists(path_usage):
+        with open(path_usage, "w") as fh:
+            write_text = "\n".join(["="*len("usage"), "Usage", "="*len("usage"), " ", ".. toctree::",
+                                    "\t :maxdepth: 3", "\t :caption: Table of Contents", " "])
+            fh.write(write_text)
+            # This can also be done using next() -> https://stackoverflow.com/a/142535/12671809 ,
+            # but reduces readability. What to do ??
+            for dirname in os.listdir(path_scripts):
+                path_dirname = os.path.join(path_scripts, dirname)
+                if os.path.isdir(path_dirname):
+                    dirname_rst = dirname + "_main"
+                    fh.write(f"\n\t {dirname}/{dirname_rst}")
+
+
+def remove_existing_docs():
+    """
+    Removes the exists documentatio in `scripts` directory
+    """
+    if os.path.exists(path_scripts):
+        shutil.rmtree(path_scripts)
+
+
+def remove_empty_dir():
+    """
+    Removes any empty directory that may have been added by mistake
+    """
+    for root, dirnames, filenames in os.walk(os.path.join(os.path.dirname(os.path.dirname(__file__)), "autometa")):
+        if os.listdir(root) == []:
+            os.rmdir(root)
```

Comment:
1. Chaining multiple methods all in one line is difficult to read. You can place the `dirpath` retrieval into a variable then pass this into `os.walk`
2. Checking an empty list with `==` is generally frowned upon compared to simply evaluating with `if not os.listdir(root)`, If the list _is_ empty, then this will evaluate to false, and if it contains elements, it will evaluate to true.

```suggestion
    pkg_dirpath = os.path.join(os.path.dirname(os.path.dirname(__file__)), "autometa")
    for root, dirnames, filenames in os.walk(pkg_dirpath):
        if not os.listdir(root):
            os.rmdir(root)
```